pkg/minikube/notify/notify.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notify
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/golang/glog"
+	"github.com/spf13/viper"
+	"k8s.io/minikube/pkg/minikube/config"
+	"k8s.io/minikube/pkg/minikube/constants"
+	"k8s.io/minikube/pkg/version"
+)
+
+const updateLinkPrefix = "https://github.com/kubernetes/minikube/releases/tag/v"
+
+var (
+	timeLayout                = time.RFC1123
+	lastUpdateCheckFilePath   = constants.MakeMiniPath("last_update_check")
+	githubMinikubeReleasesURL = "https://api.github.com/repos/kubernetes/minikube/releases"
+)
+
+func MaybePrintUpdateTextFromGithub(output io.Writer) {
+	MaybePrintUpdateText(output, githubMinikubeReleasesURL, lastUpdateCheckFilePath)
+}
+
+func MaybePrintUpdateText(output io.Writer, url string, lastUpdatePath string) {
+	if !shouldCheckURLVersion(lastUpdatePath) {
+		return
+	}
+	latestVersion, err := getLatestVersionFromURL(url)
+	if err != nil {
+		glog.Errorln(err)
+		return
+	}
+	localVersion, err := version.GetSemverVersion()
+	if err != nil {
+		glog.Errorln(err)
+		return
+	}
+	if localVersion.Compare(latestVersion) < 0 {
+		writeTimeToFile(lastUpdateCheckFilePath, time.Now().UTC())
+		fmt.Fprintln(output, fmt.Sprintf(
+			"There is a newer version of minikube available (%s%s).  Download it here:\n%s%s\n"+
```

Comment:
I tried this out and it works pretty well! One last nit, though. Could you use a multiline string here?

It will also fix a formatting issue I saw:
```shell
$ ./out/minikube start
There is a newer version of minikube available (v0.3.0).  Download it here:
https://github.com/kubernetes/minikube/releases/tag/v0.3.0
To disable this notification, add WantUpdateNotification: False to the json config file at /Users/dlorenc/.minikube/config(you may have to create the file config.json in this folder if you have no previous configuration)

Starting local Kubernetes cluster...
Kubernetes is available at https://192.168.99.100:443.
```